internal/state/execution.go

```diff
@@ -538,16 +535,8 @@ func validateValidatorUpdates(abciUpdates []abci.ValidatorUpdate,
 					pk.Type(),
 				)
 			}
-
-			if len(pk.Bytes()) != bls12381.PubKeySize {
-				return fmt.Errorf("validator %X has incorrect public key size %v",
-					valUpdate.ProTxHash, pk.String())
-			}
-
-			if pk.String() ==
-				"PubKeyBLS12381{000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}" {
-				return fmt.Errorf("validator %X public key should not be empty %v",
-					valUpdate.ProTxHash, pk.String())
+			if err := pk.Validate(); err != nil {
```

Comment:
If only one type of PubKey implements Validator, maybe it would be better to just use type casting instead of extending PubKey interface and adding mock implementations? It will be less confusing.

I mean:
```
if pubkey, ok := pk.(Validator); ok {
pubkey.Validate()
}
```